docs/api.md

```diff
@@ -71,54 +71,48 @@ _Will be removed in v3.0.0._
 
 ## Configuration
 
-### #component_parent_class
-
-Parent class for generated components
-
-    config.view_component.component_parent_class = "MyBaseComponent"
-
-Defaults to "ApplicationComponent" if defined, "ViewComponent::Base" otherwise.
-
 ### #default_preview_layout
 
 Set a custom default layout used for preview index and individual previews:
 
     config.view_component.default_preview_layout = "component_preview"
 
-### #generate_distinct_locale_files
+### #generate
 
-Always generate as many translations files as available locales:
+Configuration for generators
 
-    config.view_component.generate_distinct_locale_files = true
+Defaults to `false` unless otherwise stated.
 
-Defaults to `false`.
+Always generate a component with a sidecar directory:
 
-One file will be generated for each configured `I18n.available_locales`.
-Fallback on `[:en]` when no available_locales is defined.
+    config.view_component.generate.sidecar = true
+
+Always generate a Stimulus controller alongside the component:
 
-### #generate_locale
+    config.view_component.generate.stimulus_controller = true
 
 Always generate translations file alongside the component:
 
     config.view_component.generate_locale = true
 
-Defaults to `false`.
+Always generate as many translations files as available locales:
 
-### #generate_sidecar
+    config.view_component.generate.distinct_locale_files = true
 
-Always generate a component with a sidecar directory:
+One file will be generated for each configured `I18n.available_locales`.
+Fallback on `[:en]` when no available_locales is defined.
 
-    config.view_component.generate_sidecar = true
+Path for component files
 
-Defaults to `false`.
+    config.view_component.generate.component_path = "app/my_components"
 
-### #generate_stimulus_controller
+Defaults to `app/components`.
 
-Always generate a Stimulus controller alongside the component:
+Parent class for generated components
```

Comment:
These docs feel hard to scan, as we no longer have headings folks can skim over. How do you think we could make this new format easy to skim?

Reply:
Not sure off the top of my head, but I'll investigate.

Reply:
The `docs:build` task currently builds that `.md` file using the `mattr_accessor`s on `ViewComponent::Base`. What I think I'll start to do now is write a component that can build that markdown, both to make any future changes to the generated docs more maintainable and to provide support for any further nested options we may introduce.

Reply:
Maybe adding `###`-level headings for each option? Or displaying all these options in a table with 3 columns (option name, default value and description)?

Reply:
I went with the former about the same time as you sent this comment. It didn't actually require any changes to the implementation of our docs rake task, but I suppose it's worth keeping the changes I've made there.